### ISSUE TITLE
An error "insufficient funds for gas * price + value" is thrown

### ISSUE THREAD

**Issue by metanomad21** [ortomich/scam_with_sign#1]
When executing sendSignedTransaction, an error "insufficient funds for gas * price + value" is thrown. Can you please advise on the reason? Both userAddr and pub_addr have been confirmed to have sufficient balance.

**Comment by jmanOBO** [comment]
Yes by default the code attempts to transfer 1ETH (10e18 wei) . U should check the balance of the address to know how much to spend and also provide room for the gass price .

**Comment by jmanOBO** [comment]
> When executing sendSignedTransaction, an error "insufficient funds for gas * price + value" is thrown. Can you please advise on the reason? Both userAddr and pub_addr have been confirmed to have sufficient balance.

Yes by default the code attempts to transfer 1ETH (10e18 wei) . U should check the balance of the address to know how much to spend and also provide room for the gass price .

**Comment by unnamed-lab** [comment]
I encountered this and here was my fix:

```
const gasPrice: bigint = await web3.eth.getGasPrice();
const gasLimit: bigint = 22000n;
const txFee: bigint = BigInt(Math.floor(Number(gasPrice) * 1.666));
const funds: bigint = BigInt(amount - txFee * gasLimit);
```

From the error stated
> "insufficient funds for gas * price + value"

The funds to send is the `value - gas * gassPrice`

`const funds: bigint = BigInt(amount - txFee * gasLimit);`

This solves the error.
